### ISSUE TITLE
[Feature]: Include all IPv6 addresses in the linode_instance data source

### ISSUE THREAD

**Issue by fl1ger** [linode/terraform-provider-linode#737]
### Description

Moin!

When viewing a Linodes network setting in the WebUI it shows me all IP addresses and ranges of all protocol families. However the linode_instance data source only has the following fields:
- ip_address which has the primary linode IPv4 address without CIDR notation
- ipv6 which has the SLAAC assigned public IPv6 address with CIDR notation
- ipv4 which is a list/array/set of all the IPv4 adresses of a Linode without CIDR notation
while this itself is inconsistent I understand that for backwards compatibility is has to stay the same, but I would welcome if there would be an 
- all_ipv6 field with a list/array/set of all the IPv6 addresses/ranges configured on the machine

TIA and so long
-Ralf

### New or Affected Terraform Resources

linode_instance

### Potential Terraform Configuration

Given the following configuration:

```
resource "linode_instance" "proxy" {
        count = 1
        image = "linode/centos7"
        label = "pxy0${count.index + 1}-${var.deploymentname}"
        group = "${var.deploymentname}"
        region = "eu-central"
        type = "g6-standard-6"
        swap_size = 2048
        authorized_keys = var.linode_ssh_keys
        root_pass = var.linode_root_pw
}

resource "linode_instance_ip" "proxy1" {
  count = 3
  linode_id = linode_instance.proxy.0.id
  public = true
}

resource "linode_ipv6_range" "proxy1" {
  linode_id = linode_instance.proxy.0.id
  prefix_length = 64
}

```

it would be good to have the following snippet in the linode_instance data source:

```
    "ip_address": "139.162.148.13",
    "ipv4": [
      "139.162.148.13",
      "172.104.229.196",
      "172.104.233.83",
      "192.46.237.206"
    ],
    "ipv6": "2a01:7e01::f03c:93ff:fea2:4550/128",
    "all_ipv6": [
        "2a01:7e01::f03c:93ff:fea2:4550/128",
        "2a01:7e01:e001:385::/64"
    ]
```

**Comment by lgarber-akamai** [comment]
@fl1ger Thanks for the feature request! I think this would be a great addition to the provider.

Because the response of the [Instance Networking](https://www.linode.com/docs/api/linode-instances/#networking-information-list) endpoint is so large, it might be worth creating a `linode_instance_networking` data source to grab all of this data.

e.g. 
```
data "linode_instance_networking" "proxy" {
    id = linode_instance.proxy.id
}
```

I'll add a ticket to our internal tracker to look into this :slightly_smiling_face:

**Comment by fl1ger** [comment]
Thanks a lot. That would be great. I guess for now I have to use the API directly.

**Comment by lgarber-akamai** [comment]
This feature has been released in `v1.30.0`. Let us know if you run into any other issues 🙂
